soc/nuvoton/numaker/m46x/poweroff.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2024 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/sys/poweroff.h>
+#include <NuMicro.h>
+
+void z_sys_poweroff(void)
+{
+	SYS_UnlockReg();
+
+	/* Clear all wake-up flag */
+	CLK->PMUSTS |= CLK_PMUSTS_CLRWK_Msk;
+
+	/* Select Power-down mode */
+#if DT_NODE_HAS_PROP(DT_NODELABEL(scc), pd_mode)
```

Comment:
You could set a default in the property, then you would not need to check if the prop is set, the DT parser will always set it for you.

Reply:
Yes, to use DT_PROP_OR